### PR TITLE
Cargo-embed: fix darwin build

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change
Noticed this failure on hydra.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-embed/default.nix [hunk 1]
```diff
@@ -16,7 +23,7 @@ rustPlatform.buildRustPackage rec {
   cargoSha256 = "00p2rwqrax99kwadc8bfq8pbcijals2nzpx43wb03kwxl4955wn9";
 
   nativeBuildInputs = [ pkg-config rustfmt ];
-  buildInputs = [ libusb1 libftdi1 ];
+  buildInputs = [ libusb1 libftdi1 ] ++ lib.optionals stdenv.isDarwin [ AppKit ];
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-embed/default.nix:26]:
```suggestion
  buildInputs = [ libusb1 libftdi1 ] ++ lib.optional stdenv.isDarwin AppKit;
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12220,7 +12220,9 @@ with pkgs;
   cargo-deny = callPackage ../development/tools/rust/cargo-deny {
     inherit (darwin.apple_sdk.frameworks) Security;
   };
-  cargo-embed = callPackage ../development/tools/rust/cargo-embed { };
+  cargo-embed = callPackage ../development/tools/rust/cargo-embed {
```
Comment (figsoda) [inline pkgs/top-level/all-packages.nix:12223]:
```suggestion
  cargo-embed = callPackage ../development/tools/rust/cargo-embed {
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rust/cargo-embed/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/development/tools/rust/cargo-embed/default.nix:26]: I do like the optionals better of the optional for the simple reason that I find it more extensible. If I anything needs to be added, then it's a smaller change.
If you don't feel strongly about it, I would like to keep it.

**Comment by happysalada** [comment]
Thank you for the review! ❤️

**Review by figsoda — APPROVED** [review]
i cant test on darwin but lgtm

**Comment by happysalada** [comment]
It's a tiny fix and it won't break anything else for sure.
I don't think it's worth having more attention to it.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:12223]: Please don't hide comments and instead mark the conversation as resolved.

File: pkgs/development/tools/rust/cargo-embed/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/rust/cargo-embed/default.nix:26]: Doesn't matter really what is used. Either is fine.

**Pull request merged**